### PR TITLE
Various refactoring

### PR DESCRIPTION
**Pull request by jholdstock**
A few minor improvements made whilst working on a larger project.

These are easy to review individually and add value on their own, so I've broken them out into their own PR.

### CODE REVIEW

File: database/ticket.go [hunk 1]
```diff
@@ -91,6 +91,25 @@ type Ticket struct {
 	Outcome TicketOutcome
 }
 
+type TicketList []Ticket
```
Comment (davecgh) [inline database/ticket.go:94]: This is nothing technically incorrect about this, but I'm curious if this is a precursor to something larger?

I ask because I've personally never liked it when code does this in general without a specific need since it hides the type for no real reason.  For example, when I'm looking at the code and am not already familiar with it and I see:

```Go
func (vdb *VspDatabase) GetUnconfirmedTickets() (TicketList, error) {
```

I think, hmm, ok, now I have to go figure out what `TicketList` is.  Is it a map?  A slice?  A generic container? Something else?

In other words, I generally find that it adds more cognitive overhead in general.

File: rpc/dcrd.go [hunk 2]
```diff
@@ -228,9 +228,19 @@ func (c *DcrdRPC) GetBestBlockHeader() (*dcrdtypes.GetBlockHeaderVerboseResult,
...
+	if err != nil {
+		return nil, err
+	}
+	return blockHeader, nil
+}
+
+// GetBlockHeaderVerbose uses getblockheader RPC with verbose=true to retrieve
+// the header of the requested block.
+func (c *DcrdRPC) GetBlockHeaderVerbose(blockHash string) (*dcrdtypes.GetBlockHeaderVerboseResult, error) {
 	verbose := true
```
Comment (davecgh) [inline rpc/dcrd.go:241]: No need for a local variable here.

Suggest:

```Go
const verbose = true
```

**Review by dajohi — APPROVED** [review]

File: database/ticket.go (hunk 1 above)
Comment (jholdstock) [reply database/ticket.go:94]: Interesting point I hadn't considered before. I do anticipate adding more funcs to TicketList later, but nothing imminent. 

It's mainly just to make the code calling `EarliestPurchaseHeight` cleaner, but also has the side-effect of making autocomplete more useful. Probably a habit from my (mostly) OOP background.

File: rpc/dcrd.go (hunk 2 above)
Comment (jholdstock) [reply rpc/dcrd.go:241]: Good call, theres a few more of those to update as well.

**Review by davecgh — APPROVED** [review]

File: database/ticket.go (hunk 1 above)
Comment (davecgh) [reply database/ticket.go:94]: Gotcha.  Like I said, nothing wrong with it technically.  I just figured I'd offer my perspective since I was reviewing.
